src/test/java/xades4j/production/UncheckedSignerBESTest.java

```diff
@@ -120,16 +132,32 @@ public void testUncheckedSignBes() throws Exception
     public void testUncheckedSignBesGood() throws Exception
     {
         System.out.println("uncheckedSignBesGood");
-        CertificateValidationProvider prov = genValidationProvider("unchecked/TestCA.cer", "unchecked");
-        trySignAndVerify(keyingProviderGood, prov, "document.unchecked.signed.bes.good.xml");
+        trySignAndVerify(keyingProviderGood, validationProvider, "document.unchecked.signed.bes.good.xml");
     }
 
     @Test
     public void testUncheckedSignBesNoSignKeyUsage() throws Exception
     {
         System.out.println("uncheckedSignBesNoSignKeyUsage");
-        CertificateValidationProvider prov = genValidationProvider("unchecked/TestCA.cer", "unchecked");
-        // BUG: Validation passes even though the keyUsage is invalid!
-        trySignAndVerify(keyingProviderNoSign, prov, "document.unchecked.signed.bes.nosign.xml");
+
+        try
+        {
+            trySignAndVerify(keyingProviderNoSign, validationProvider, "document.unchecked.signed.bes.nosign.xml");
+            fail("expected SigningCertificateKeyUsageException did not occur");
```

Comment:
I'd prefer to use `assertThrows()` instead of try/catch and `fail()`, but the `pom.xml` requires Java 7 support.

Reply:
You can use `@Test(expected = SigningCertificateKeyUsageException.class)` and remove the try/catch. Check for other usages in the codebase. Can you update?

Reply:
Done in ab27d62576c21ecc2da53dc5071c0573bbb9a46a.